util/rowDecoder/decoder.go

```diff
@@ -38,17 +38,24 @@ type Column struct {
 
 // RowDecoder decodes a byte slice into datums and eval the generated column value.
 type RowDecoder struct {
-	tbl         table.Table
-	mutRow      chunk.MutRow
-	colMap      map[int64]Column
-	colTypes    map[int64]*types.FieldType
-	defaultVals []types.Datum
-	cols        []*table.Column
-	pkCols      []int64
+	tbl table.Table
+	// mutRow is used to evaluate the virtual generated column.
+	mutRow          chunk.MutRow
+	needMutRow      bool
+	colMap          map[int64]Column
+	nonGCColMap     map[int64]Column
+	gcColMap        map[int64]Column
+	orderedGCOffset []int
+	offset2Id       map[int]int
```

Comment:
```suggestion
	offset2Id       map[int]int64
```

We can use int64 so that no type conversion is needed.